src/ClientData/ProcessDataTest.cpp

```diff
@@ -187,6 +188,52 @@ TEST(ProcessData, IsModuleLoaded) {
   EXPECT_FALSE(process.FindModuleByPath("not/loaded/module").has_value());
 }
 
+TEST(ProcessData, IsModuleLoaded) {
+  ModuleInfo module_info_1;
+  module_info_1.set_file_path("path/to/file1");
+  module_info_1.set_address_start(0);
+  module_info_1.set_address_end(10);
+
+  ModuleInfo module_info_2;
+  module_info_2.set_file_path("path/to/file2");
+  module_info_2.set_address_start(100);
+  module_info_2.set_address_end(110);
+  module_info_2.set_build_id("build_id_2");
+
+  ProcessData process(ProcessInfo{});
+  process.UpdateModuleInfos({module_info_1, module_info_2});
+
+  // path empty
+  EXPECT_FALSE(process.IsModuleLoaded(""));
+
+  // wrong path
+  EXPECT_FALSE(process.IsModuleLoaded("/path/to/file1"));
```

Comment:
The function `IsModuleLoaded(std::string file_path)` was already part of `ProcessData` before I made the commit. I added the overload `IsModuleLoaded(ModuleData* module)`. It is an overload (and not a different method) because these functions do the same thing. Check whether a module was loaded by the process. They are slightly different, the second variant also checks the `build_id`. I hope that explains it